File: CS/ClientChartDesignerSample/SessionHelper.cs
using System;
using System.Web.SessionState;
using DevExpress.XtraCharts;
using DevExpress.XtraCharts.Web;

namespace ClientChartDesignerSample {
    public static class SessionHelper {
        const string chartLayoutName = "ChartLayout";

        public static void SaveChartLayout(String layoutXml, HttpSessionState session) {
            session[chartLayoutName] = layoutXml;
        }

        public static void SaveChart(WebChartControl chart, HttpSessionState session) {
            SaveChartLayout(chart.SaveToXml(), session);
        }

        public static void LoadChart(WebChartControl chart, HttpSessionState session) {
            if (session[chartLayoutName] != null) {
                String layoutXml = session[chartLayoutName] as String;
                if (layoutXml != null)
                    chart.LoadFromXml(layoutXml);
            }
            else
                InitSeries(chart);
        }

        static void InitSeries(WebChartControl chart) {
            using (NWindEntities dbContext = new NWindEntities()) {
                SideBySideBarSeriesView view = new SideBySideBarSeriesView();
                view.FillStyle.FillMode = FillMode.Solid;
                Series series = new Series() {
                    Name = "Product Price",
                    View = view
                };
                series.ArgumentDataMember = "ProductName";
                series.ValueDataMembers.AddRange(new string[] { "UnitPrice" });
                chart.Series.Add(series);
            }
        }
    }
}

File: VB/ClientChartDesignerSample/SessionHelper.vb
Imports DevExpress.XtraCharts.Web
Imports DevExpress.XtraCharts

Public NotInheritable Class SessionHelper
    Const chartLayoutName As String = "ChartLayout"

    Public Shared Sub SaveChartLayout(layoutXml As String, session As HttpSessionState)
        session(chartLayoutName) = layoutXml
    End Sub

    Public Shared Sub SaveChart(chart As WebChartControl, session As HttpSessionState)
        SaveChartLayout(chart.SaveToXml(), session)
    End Sub

    Public Shared Sub LoadChart(chart As WebChartControl, session As HttpSessionState)
        If (session(chartLayoutName) IsNot Nothing) Then
            Dim layoutXml As String = TryCast(session(chartLayoutName), String)
            If (layoutXml IsNot Nothing) Then
                chart.LoadFromXml(layoutXml)
            End If
        Else
            InitSeries(chart)
        End If

    End Sub

    Public Shared Sub InitSeries(chart As WebChartControl)
        Using dbContext As New NWindEntities()
            Dim view As SideBySideBarSeriesView = New SideBySideBarSeriesView()
            view.FillStyle.FillMode = FillMode.Solid
            Dim series As Series = New Series()
            series.Name = "Product Price"
            series.View = view
            series.ArgumentDataMember = "ProductName"
            series.ValueDataMembers.AddRange(New String() {"UnitPrice"})
            chart.Series.Add(Series)
        End Using
    End Sub
End Class
